### PR TITLE
Some changes

### PR DESCRIPTION
**Pull request by NicolasAlmerge**
- Changed doc with better formatting and up to date
- Added variables `tau`, `ϕ` and `phi`
- Removed unsafe `hypot` function (done without the `safe_pow` function). The `nhypot` function is now called `hypot`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,18 +1,22 @@
 # plusminus
 
-The **plusminus** package provides a ready-to-run arithmetic parser and evaluator, based on `pyparsing`'s 
-`infixNotation` helper method.
+The **plusminus** package provides a ready-to-run arithmetic parser and evaluator, based on [`pyparsing`](https://github.com/pyparsing/pyparsing)'s 
+[`infixNotation`](https://github.com/pyparsing/pyparsing/blob/master/pyparsing/helpers.py#L608) helper method.
```
Comment (ptmcg) [inline README.md]: Links to code by line are probably pretty fragile - better to link to docs at `pyparsing-doc.readthedocs`

File: plusminus/plusminus.py [hunk 2]
```diff
@@ -1094,6 +1094,9 @@ def vars(self):
         return ret
 
 
+phi = (1.0 + 5**0.5) / 2.0   # The golden number
```
Comment (ptmcg) [inline plusminus/plusminus.py]: Does not need to be defined at module scope.

File: doc/example_parsers.md [hunk 3]
```diff
@@ -11,37 +11,39 @@
...
+
+        def fv(pv, rate, n_periods):
+            return pv * safe_pow(1 + rate, n_periods)
+
+        def pp(pv, rate, n_periods):
+            return rate * pv / (1 - safe_pow(1 + rate, -n_periods))
+
+        super().customize()
+        self.add_operator("of", 2, ArithmeticParser.LEFT, lambda a, b: a * b)
+        self.add_operator('%', 1,ArithmeticParser.LEFT, lambda a: a / 100)
```
Comment (ptmcg) [inline doc/example_parsers.md]: Need a space after ','

File: plusminus/plusminus.py [hunk 4]
```diff
@@ -1134,8 +1140,7 @@ def customize(self):
             (lambda a, b: int(abs(a) / math.gcd(a, b) * abs(b)) if a or b else 0),
         )
         self.add_function("gamma", 2, math.gamma)
-        self.add_function("hypot", 2, math.hypot)
-        self.add_function("nhypot", ..., lambda *seq: sum(safe_pow(i, 2) for i in seq)**0.5)
+        self.add_function("hypot", ..., lambda *seq: sum(safe_pow(i, 2) for i in seq)**0.5)
```
Comment (ptmcg) [inline plusminus/plusminus.py:1142]: Thanks, I was meaning to do this too.

File: plusminus/plusminus.py [hunk 5]
```diff
@@ -1109,9 +1112,12 @@ def customize(self):
         super().customize()
         self.initialize_variable("pi", math.pi)
         self.initialize_variable("π", math.pi)
-        self.initialize_variable("τ", math.pi * 2)
+        self.initialize_variable("τ", math.tau)
+        self.initialize_variable("tau", math.tau)
         self.initialize_variable("e", math.e)
-        self.initialize_variable("φ", (1 + 5 ** 0.5) / 2)
+        self.initialize_variable("φ", phi)
+        self.initialize_variable("ϕ", phi)
```
Comment (ptmcg) [inline plusminus/plusminus.py:1118]: Are capital and lower-case phi's needed?

File: plusminus/plusminus.py (hunk 2 above)
Comment (ptmcg) [inline plusminus/plusminus.py]: Not needed at module scope.

File: doc/arithmetic_parser.md [hunk 6]
```diff
@@ -165,57 +165,63 @@ functions, and common mathematical variables
...
-                self.add_operator(
-                    factorial_operator, 1, ArithmeticParser.LEFT, constrained_factorial
-                )
-                self.add_operator("⁻¹", 1, ArithmeticParser.LEFT, lambda x: 1 / x)
-                self.add_operator("²", 1, ArithmeticParser.LEFT, lambda x: safe_pow(x, 2))
-                self.add_operator("³", 1, ArithmeticParser.LEFT, lambda x: safe_pow(x, 3))
-                self.add_operator("√", 1, ArithmeticParser.RIGHT, lambda x: x ** 0.5)
-                self.add_operator("√", 2, ArithmeticParser.LEFT, lambda x, y: x * y ** 0.5)
+    ```python
+    phi = (1.0 + 5**0.5) / 2.0   # The golden number
```
Comment (ptmcg) [inline doc/arithmetic_parser.md]: Not at module scope, move to BasicArithmeticParser.customize

**Review by ptmcg — CHANGES_REQUESTED** [review]

**Comment by TheOneMusic** [comment]
@ptmcg I added the ability for maintainers to modify the `beta` branch of my repository

File: plusminus/plusminus.py (hunk 5 above)
Comment (NicolasAlmerge) [reply plusminus/plusminus.py:1118]: Yes, take a look at this:

```python
>>> "ϕ".lower()
'ϕ'
>>> _.upper()
'Φ'
>>> _.lower()
'φ'
```
Basically `φ` and `ϕ` are two lower-case characters, while `Φ` is the upper-case of those two

Comment (ptmcg) [reply plusminus/plusminus.py:1118]: Interesting, TIL there are 2 different lower case PHI unicode symbols, one is the actual lowercase, the other is a compatibility "old-style" character for mathematical use. Ok, that's great, define them both.

Described here https://en.wikipedia.org/wiki/Phi#Computing

**Comment by TheOneMusic** [comment]
@ptmcg Are there other things to change?

**Comment by TheOneMusic** [comment]
@ptmcg Is everything okay?

**Comment by TheOneMusic** [comment]
@ptmcg Is everything okay now?

**Pull request merged**